lib/util/pools.js

```diff
@@ -78,10 +78,8 @@ export function createPool(name, opts) {
       }
     },
 
-    freeAll: function() {
-      let allocatedLength = allocated.length;
-      let freeLength = free.length;
-      let reAlloc = allocated.slice(0, size - freeLength);
+    freeAll: function freeAllPool() {
```

Comment:
I recommend switching to:

`freeAll() {}`

rather than trying to name the fn's yourself. In large codebases I find that manually naming things is one of the first things to go unmaintained. Over time I end up seeing:

```js
{
  methodOne: function methodOne() {},
  methodTwo: function methodOne() {},
  methodThree: function methodOne() {}
}
```

which is understandable, 'cause it's so easy to miss!

In an ES2015-compliant environment, even anonymous functions will be automatically named ([spec here](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-object-initializer-runtime-semantics-propertydefinitionevaluation)), but this isn't implemented in any environment that I know of atm. What **is** implemented is giving a name to a function defined with the shorthand syntax, like:

```js
freeAll() {}
```

You can test this by running these quick tests in Chrome:

```js
var testOne = {
  someProp: function() {}
};
testOne.someProp.name === ""; // true
```

```js
var testTwo = {
  someProp() {}
};
testTwo.someProp.name === "someProp"; // true
```